src/util.jl

```diff
@@ -148,6 +148,28 @@ function slice(dims::NTuple{N}, i, j, low = 1, trim = 0) where N
     CartesianIndices(ntuple(k-> k == j ? (i:i) : (low:dims[k] - trim), N))
 end
 
+# """
+#     bc_indices(Ng)
+
+# Given an array size Ng = (N, M, ...), that includes the ghost cells, it returns a
+# Vector of Tuple(s) in which each Tuple is composed of a ghost cell CartesianIndex,
+# its respective donor cell CartesianIndex, and the normal direction between them:
+# [Tuple{CartesianIndex, CartesianIndex, Int}, ...]
+# """
+# function bc_indices(Ng)
+#     D = length(Ng)
+#     bc_list = Tuple{CartesianIndex, CartesianIndex, Int}[]
+#     for d ∈ 1:D
+#         slice_ghost_start = slice(Ng, 0, d, 1, 2)
+#         slice_donor_start = slice_ghost_start .+ δ(d, D)
```

Comment:
Is this correct for both scalars and vectors?

Reply:
This just gets you the list of indices for ghost-donor cells, which should be the same for both scalar and vectors AFAIK.
Then the imposition of the boundary conditions is specialized for each of them in `function BC!(u, U, bc, f = 1.0)` and `BC!(u, bc)`.

Reply:
Yes, that's true.